### PR TITLE
Data race fix.

### PR DESCRIPTION
**Pull request by micmich**

### CODE REVIEW

File: com.dynatrace.eclipseintegration.plugin/src/com/dynatrace/diagnostics/launcher/LaunchTerminated.java [hunk 1]
```diff
@@ -157,11 +157,21 @@ private long timeoutMs() {
...
+				// The user waits on a progress bar anyhow.
+
+				for (WeakReference<ILaunch> iLaunchWeakReference : JUnitLaunch_testCount.keySet()) {
+					ILaunch launchDereferenced = iLaunchWeakReference.get();
+					if (launch.equals(launchDereferenced)) {
+						return JUnitLaunch_testCount.get(iLaunchWeakReference);
+					}
 				}
+				Thread.sleep(100);
+				System.out.println("Ping");
```
Comment (dglugla) [inline com.dynatrace.eclipseintegration.plugin/src/com/dynatrace/diagnostics/launcher/LaunchTerminated.java]: leftover?

File: com.dynatrace.eclipseintegration.plugin/src/com/dynatrace/diagnostics/launcher/LaunchTerminated.java [hunk 2]
```diff
@@ -157,11 +157,21 @@ private long timeoutMs() {
 
 	private final TestRunRecorder.ContainsLaunchTestCounts launchTestCounts = new TestRunRecorder.ContainsLaunchTestCounts() {
 		@Override
-		public int expectedTestCountsFor(ILaunch launch) {
-			for (WeakReference<ILaunch> iLaunchWeakReference : JUnitLaunch_testCount.keySet()) {
-				if (launch.equals(iLaunchWeakReference.get())) {
-					return JUnitLaunch_testCount.get(iLaunchWeakReference);
+		public int expectedTestCountsFor(ILaunch launch) throws InterruptedException {
+			long startPolling = System.currentTimeMillis();
+			while (System.currentTimeMillis() < startPolling + 30*1000 ) {
```
Comment (dglugla) [inline com.dynatrace.eclipseintegration.plugin/src/com/dynatrace/diagnostics/launcher/LaunchTerminated.java]: extract "30*1000" to a constant with a self-explainable name

File: com.dynatrace.eclipseintegration.plugin/src/com/dynatrace/diagnostics/launcher/LaunchTerminated.java [hunk 3]
```diff
@@ -157,11 +157,21 @@ private long timeoutMs() {
...
+				// to be available within a sensible time span, normally arrives within 1 second.
+				// The user waits on a progress bar anyhow.
+
+				for (WeakReference<ILaunch> iLaunchWeakReference : JUnitLaunch_testCount.keySet()) {
+					ILaunch launchDereferenced = iLaunchWeakReference.get();
+					if (launch.equals(launchDereferenced)) {
+						return JUnitLaunch_testCount.get(iLaunchWeakReference);
+					}
 				}
+				Thread.sleep(100);
```
Comment (dglugla) [inline com.dynatrace.eclipseintegration.plugin/src/com/dynatrace/diagnostics/launcher/LaunchTerminated.java]: extract sleep time to a constant

File: com.dynatrace.eclipseintegration.plugin/src/com/dynatrace/diagnostics/launcher/LaunchTerminated.java [hunk 4]
```diff
@@ -157,11 +157,21 @@ private long timeoutMs() {
...
 	private final TestRunRecorder.ContainsLaunchTestCounts launchTestCounts = new TestRunRecorder.ContainsLaunchTestCounts() {
 		@Override
-		public int expectedTestCountsFor(ILaunch launch) {
-			for (WeakReference<ILaunch> iLaunchWeakReference : JUnitLaunch_testCount.keySet()) {
-				if (launch.equals(iLaunchWeakReference.get())) {
-					return JUnitLaunch_testCount.get(iLaunchWeakReference);
+		public int expectedTestCountsFor(ILaunch launch) throws InterruptedException {
+			long startPolling = System.currentTimeMillis();
+			while (System.currentTimeMillis() < startPolling + 30*1000 ) {
+				// Required data arrives from a different thread. I expect the data
```
Comment (dglugla) [inline com.dynatrace.eclipseintegration.plugin/src/com/dynatrace/diagnostics/launcher/LaunchTerminated.java]: use indirect form in sentences; treat the code as public, so if you need to leave a comment make it impersonal

**Review by dglugla — CHANGES_REQUESTED** [review]

File: com.dynatrace.eclipseintegration.plugin/src/com/dynatrace/diagnostics/launcher/LaunchTerminated.java (hunk 1 above)
Comment (micmich) [reply com.dynatrace.eclipseintegration.plugin/src/com/dynatrace/diagnostics/launcher/LaunchTerminated.java]: OMG

**Review by dglugla — APPROVED** [review]

**Pull request merged**